### PR TITLE
Preset queue endpoint

### PR DESCRIPTION
**Pull request by bvanwijk**
This contains all the endpoints to change/update/edit the presetqueues.
Let me know what you think about it, or if I have missed something.

### CODE REVIEW

File: backend/src/main/java/com/benine/backend/http/presetqueue/CreatePresetQueueHandler.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    if (name != null) {
+      PresetQueue presetQueue = new PresetQueue(name, new ArrayList<>());
+    
+      getPresetQueueController().addPresetQueue(presetQueue);
+    
+      respondSuccess(request, res);
+      getLogger().log("New preset Queue created", LogEvent.Type.INFO);
+    } else {
+      respondFailure(request, res);
+      getLogger().log("New preset can't be created", LogEvent.Type.WARNING);
```
Comment (jochem725) [inline backend/src/main/java/com/benine/backend/http/presetqueue/CreatePresetQueueHandler.java]: I think you mean preset queue here.

**Comment by jochem725** [comment]
Looks great, I think you have everything covered.

File: backend/src/main/java/com/benine/backend/http/presetqueue/CreatePresetQueueHandler.java (hunk 1 above)
Comment (bvanwijk) [inline backend/src/main/java/com/benine/backend/http/presetqueue/CreatePresetQueueHandler.java]: Well noticed, I have fixed it.

File: backend/src/main/java/com/benine/backend/http/presetqueue/AddPresetToPresetQueueHandler.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  /**
+   * Constructs a handler for adding a preset to a preset queue /presetqueues/{id}/addpreset
+   * @param httpserver this handler belongs to.
+   */
+  public AddPresetToPresetQueueHandler(HTTPServer httpserver) {
+    super(httpserver);
+  }
+
+  @Override
+  public void handle(String arg0, Request request, HttpServletRequest arg2, HttpServletResponse res)
```
Comment (NaomidR) [inline backend/src/main/java/com/benine/backend/http/presetqueue/AddPresetToPresetQueueHandler.java]: Maybe you can split this method into smaller methods.

File: backend/src/main/java/com/benine/backend/http/presetqueue/EditPresetQueueHandler.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    
+    if (presetQueue != null && name != null) {
+      presetQueue.setName(name);
+      getPresetQueueController().updatePresetQueue(presetQueue);
+      respondSuccess(request, res);
+      getLogger().log("The name of Preset queue " + id 
+                                + " is succesfully updated.", LogEvent.Type.INFO);
+    } else {
+      respondFailure(request, res);
+      getLogger().log("Preset queue " + id + "is not deleted.", LogEvent.Type.WARNING);
```
Comment (NaomidR) [inline backend/src/main/java/com/benine/backend/http/presetqueue/EditPresetQueueHandler.java]: Small comment, but before "is not deleted" there should be a space I think.

**Comment by NaomidR** [comment]
I placed two small comments but if you've fixed them I agree with merging.

**Comment by bvanwijk** [comment]
@NaomidR I've fixed your comments.

**Comment by Kingdorian** [comment]
Looks really good, I think this can be merged.

**Comment by egedebruin** [comment]
Very good work, I agree with merging.

**Comment by jochem725** [comment]
Let's merge it then 😄

**Pull request merged**